### PR TITLE
Fs

### PR DESCRIPTION
**Pull request by pkivolowitz**
This includes much of the last pull request, which I cancelled due to incompleteness. 

Some of these commits are minor because I am editing on an I7 with sublime but testing on the ARM. The I7, being in a VM, has unreliable access to USB DACs.

### CODE REVIEW

File: curses/client.cpp [hunk 1]
```diff
@@ -561,29 +565,35 @@ int main(int argc, char * argv[])
...
-			DisplayTracks();
-			wmove(top_window, 1, 1);
-			wrefresh(instruction_window);
-			wrefresh(top_window);
-			wrefresh(bottom_window);
-			wrefresh(mid_left);
-			wrefresh(mid_right);
+			// THESE ARE HAPPENING TOO FREQUENTLY.
+			if (display_needs_update || difftime(time(nullptr), last_update) > 0.2) {
+				DACInfoCommand();
```
Comment (LOZORD) [inline curses/client.cpp]: I would put all of these function calls into a wrapper function or something like `onUpdate`.

File: fsdev/db.cpp [hunk 2]
```diff
@@ -143,38 +131,35 @@ bool InsertTrack(sql::Connection * connection, DIRENT * d, string & nspace, Logg
...
+		if (stmt == nullptr) {
+			throw LOG2(_log_, "prepareStatement() failed", LogLevel::FATAL);
 		}
 		stmt->setString(1, to_string(d->up));
 		stmt->setString(2, d->name);
 		stmt->setString(3, nspace);
 
-		for (size_t i = 3; i < n; i++)
-		{
+		for (size_t i = 3; i < n; i++) {
```
Comment (LOZORD) [inline fsdev/db.cpp]: What is the significance of `3` here?

File: protos/commands.proto [hunk 3]
```diff
@@ -10,6 +10,7 @@ enum LogLevel {
 	MINIMAL = 1;
 	CONVERSATIONAL = 2;
 	VERBOSE = 3;
+	REDICULOUS = 4;
```
Comment (LOZORD) [inline protos/commands.proto:13]: `RIDICULOUS`, or even better `LUDICROUS` :wink:

**Review by LOZORD — APPROVED** [review]

File: fsdev/db.cpp (hunk 2 above)
Comment (pkivolowitz) [reply fsdev/db.cpp]: Note above the magical 3, there are three setStrings. These three special ones come from the "file system structure" being built by this program. The remaining fields come from ffprobe. Adding comments now.

File: protos/commands.proto (hunk 3 above)
Comment (pkivolowitz) [reply protos/commands.proto:13]: Uggh. Let the misspelling be a artifact.

**Pull request merged**